File: BilDunya/src/main/java/com/bildunya/dto/SendChatMessageRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SendChatMessageRequest {

    @NotBlank(message = "Receiver username is required")
    private String receiverUsername;

    @NotBlank(message = "Text is required")
    @Size(max = 2000, message = "Text must be at most 2000 characters")
    private String text;
}

File: bildunya_frontend/lib/data/models/send_chat_message_request.dart
/// [com.bildunya.dto.SendChatMessageRequest]
class SendChatMessageRequest {
  const SendChatMessageRequest({
    required this.receiverUsername,
    required this.text,
  });

  final String receiverUsername;
  final String text;

  Map<String, dynamic> toJson() => {
        'receiverUsername': receiverUsername,
        'text': text,
      };
}
